Update watchOS deployment target to 3.0

sushichop opened the pull request:
### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: #954 

### Pull Request Description

I updated watchOS deployment target to 3.0 and added `DDOSLoggerTests` that is almost the same as( a little different from) `DDBasingLoggingTests`. 
And I also added a few tests to check `fileName` property in DDLogMessge.

lolgear commented:
You have copy-pasted tests... For what reason?
It is a bad style.

sushichop commented:
It is just a test for check logging(DDLogMesage) by DDOSLogger. 
A better test or style is welcome:)

ffried commented:
I think we should add (legacy) tests for `DDTTYLogger` and `DDASLLogger` and instead change the logger used in `DDBasicLoggingTests` to `DDOSLogger`.

lolgear commented:
@ffried,
`DDBasicLoggingTests` could cover whole set of default loggers (TTY, ASL, OS).

bpoplauschi commented on Classes/DDASLLogCapture.h:
```diff
@@ -20,6 +20,7 @@
 /**
  *  This class provides the ability to capture the ASL (Apple System Logs)
  */
+API_DEPRECATED("Use DDOSLogger instead", macosx(10.4,10.12), ios(2.0,10.0), watchos(2.0,3.0), tvos(9.0,10.0))
```
Nice

bpoplauschi commented on Classes/DDASLLogCapture.m:
```diff
@@ -20,6 +20,8 @@
     #define DD_LEGACY_MACROS 0
 #endif
 
+#if !TARGET_OS_WATCH
```
👍

bpoplauschi commented on Classes/DDASLLogger.h:
```diff
@@ -41,6 +41,7 @@ extern const char* const kDDASLDDLogValue;
  * However, if you instead choose to use file logging (for faster performance),
  * you may choose to use a file logger and a tty logger.
  **/
+API_DEPRECATED("Use DDOSLogger instead", macosx(10.4,10.12), ios(2.0,10.0), watchos(2.0,3.0), tvos(9.0,10.0))
```
👍

OSSPolice commented:
<!--
  0 Errors
  2 Warnings: Any changes to library code sh...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any changes to library code should be reflected in the CHANGELOG. Please consider adding a note there about your change.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">The CocoaPods podspec was modified but the Carthage project wasn’t. Did you forget to update the xcodeproj?</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bpoplauschi commented on Configs/Module-Shared.xcconfig:
```diff
@@ -228,4 +228,4 @@ TVOS_DEPLOYMENT_TARGET = 9.0
 
 // Code will load on this and later versions of watchOS. Framework APIs that are unavailable in earlier versions will be weak-linked; your code should check for null function pointers or specific system versions before calling newer APIs.
-WATCHOS_DEPLOYMENT_TARGET = 2.0
+WATCHOS_DEPLOYMENT_TARGET = 3.0
```
That's why we have `xcconfig` - to be able to change that setting in one place only

lolgear replied:
This is ~~(the only one true way)~~ dirty hack in monolithic environment in xcode project. :) 
Xcode can't copy targets well, it also can't share build configurations and many more "monolithic" issues.

sushichop commented:
Yes, but I think DDBasicLoggingTests almost covers DDTTYLogger, DDASLogger, and DDOSLogger.
Then this time I added test for using DDOSLogger.

sushichop commented:
> `DDBasicLoggingTests` could cover whole set of default loggers (TTY, ASL, OS).

Yes. I think so too.

sushichop commented:
So if we add tests for DDTTYLogger and DDASLogger, I think it is enough to add diffs for each Logger
(for example, by creating another PR).

bpoplauschi commented:
I agree with @ffried : for`DDBasicLoggingTests` just replace `DDTTYLogger` with `DDOSLogger`. Since we are deprecating `DDASLLogger`, I don't really care about testing it.

Should we also reprecate `DDTTYLogger`?

lolgear commented:
@bpoplauschi 
In which version will you obsolete all loggers except DDOSLogger?

sushichop commented:
> I agree with @ffried : for`DDBasicLoggingTests` just replace `DDTTYLogger` with `DDOSLogger`. Since we are deprecating `DDASLLogger`, I don't really care about testing it.
> 
> Should we also reprecate `DDTTYLogger`?

Now I'm a little confused...😅 Do you say it is enough to replace `DDTYLogger` with `DDOSLogger`?

bpoplauschi commented:
@lolgear :) man, I was asking a question, what do you all think about marking as deprecated `DDASLLogger` and `DDTTYLogger`.
In my opinion, we can mark them for the next version (3.5.0) as this is not breaking backwards compatibility.

@sushichop yes, I mean for the `DDBasicLoggingTests` we can just replace with `DDOSLogger`. The point of the `DDBasicLoggingTests` is to test the overall mechanism from log macro (i.e. `DDLogDebug`) to any logger, not necessarily one of them.
If we want to add specific logger tests, we can do so, but I would not spend time in adding tests for classes we want to deprecate.

sushichop commented:
I updated commit. 
Sorry guys, it's time over for me today(I'll take a bath and go to bed). Good night!
Anyway I'm glad to discuss various things recently:)

Thanks in advance.

sushichop commented:
Hi guys! Finally I updated all commits. I summarize my opinion at the moment.
See each commit for details:)

- I left DDBasicLoggingTests.m using `DDTTYLogger`. Because I think it is necessary to test all OS version(e.g. iOS 8.0 to 12.0) and overall mechanism from log macro.
- I added DDOSLoggingTests.m using DDOSLogger and DDASLogger.
  (e.g. DDOSLogger for iOS 10.0+ and DDASLogger for iOS 10.0-)
- I added iOS 9.3 simulator for testing DDASLogger using travis-ci.

If you think tests for DDASLogger is unnecessary, I'll gladly remove it.

That's all. Thanks!
